File: Step 01 - Basic/Basic_Hashing/Find the highest or lowest frequency element using Brute Force Approach.js
class FrequencyCounter {
    countFreq(arr) {
        const n = arr.length;
        const visited = new Array(n).fill(false); // Track if element is already processed
        let maxFreq = 0, minFreq = n;             // Initialize min and max frequency
        let maxEle = 0, minEle = 0;               // Initialize elements

        for (let i = 0; i < n; i++) {

            // Skip already processed element
            if (visited[i]) continue;

            // Count how many times arr[i] appears
            let count = 1;
            for (let j = i + 1; j < n; j++) {
                if (arr[i] === arr[j]) {
                    visited[j] = true;  // Mark this index as processed
                    count++;
                }
            }

            // Check if this element has max frequency
            if (count > maxFreq) {
                maxEle = arr[i];
                maxFreq = count;
            }

            // Check if this element has min frequency
            if (count < minFreq) {
                minEle = arr[i];
                minFreq = count;
            }
        }

        // Output the result
        console.log("The highest frequency element is:", maxEle);
        console.log("The lowest frequency element is:", minEle);
    }
}

// Create an instance and run
const fc = new FrequencyCounter();
const arr = [10, 5, 10, 15, 10, 5]; // Sample input
fc.countFreq(arr);                 // Call method to compute frequencies

File: Step 01 - Basic/Basic_Hashing/Find the highest or lowest frequency element using Brute Force Approach.c
#include <stdio.h>
#include <stdbool.h>

void countFreq(int arr[], int n) {
    bool visited[n];   // Track visited elements

    // Initialize visited array
    for (int i = 0; i < n; i++) {
        visited[i] = false;
    }

    int maxFreq = 0;
    int minFreq = n;
    int maxEle = 0;
    int minEle = 0;

    for (int i = 0; i < n; i++) {

        // Skip already processed elements
        if (visited[i] == true)
            continue;

        int count = 1;

        // Count frequency of arr[i]
        for (int j = i + 1; j < n; j++) {
            if (arr[i] == arr[j]) {
                visited[j] = true;
                count++;
            }
        }

        // Update max frequency
        if (count > maxFreq) {
            maxFreq = count;
            maxEle = arr[i];
        }

        // Update min frequency
        if (count < minFreq) {
            minFreq = count;
            minEle = arr[i];
        }
    }

    printf("The highest frequency element is: %d\n", maxEle);
    printf("The lowest frequency element is: %d\n", minEle);
}

int main() {
    int arr[] = {10, 5, 10, 15, 10, 5};
    int n = sizeof(arr) / sizeof(arr[0]);

    countFreq(arr, n);

    return 0;
}